crates/services/sync/src/import.rs

```diff
@@ -274,105 +301,109 @@ fn get_block_stream<
     params: &Config,
     p2p: Arc<P>,
     consensus: Arc<C>,
-) -> impl Stream<Item = impl Future<Output = anyhow::Result<Option<(PeerId, SealedBlock)>>>>
-{
-    get_header_stream(range, params, p2p.clone()).map({
-        let p2p = p2p.clone();
-        let consensus_port = consensus.clone();
-        move |batch| {
-            {
-                let p2p = p2p.clone();
-                let consensus_port = consensus_port.clone();
-                get_sealed_blocks(batch, p2p.clone(), consensus_port.clone())
+) -> impl Stream<Item = impl Future<Output = SealedBlockBatch>> + '_ {
+    let header_stream = get_header_batch_stream(range.clone(), params, p2p.clone());
+    header_stream
+        .map({
+            let consensus = consensus.clone();
+            let p2p = p2p.clone();
+            move |header_batch: SealedHeaderBatch| {
+                let Batch {
+                    peer,
+                    range,
+                    results,
+                } = header_batch;
+                let checked_headers = results
+                    .into_iter()
+                    .take_while(|header| {
+                        check_sealed_header(header, peer.clone(), &p2p, &consensus)
+                    })
+                    .collect::<Vec<_>>();
+                Batch::new(peer, range, checked_headers)
+            }
+        })
+        .map(move |headers| {
+            let consensus = consensus.clone();
+            let p2p = p2p.clone();
+            async move {
+                let Batch {
+                    peer,
+                    range,
+                    results,
+                } = headers;
+                if results.is_empty() {
+                    SealedBlockBatch::new(peer, range, vec![])
+                } else {
+                    await_da_height(
+                        results
+                            .last()
+                            .expect("We checked headers are not empty above"),
+                        &consensus,
+                    )
+                    .await;
+                    let headers =
+                        SealedHeaderBatch::new(peer.clone(), range.clone(), results);
```

Comment:
Seems weird that these need to be `.clone()`.

Reply:
It looks like I was able to remove these clones without issue.